src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -996,6 +1041,38 @@ void Esp32WiFiManager::enable_esp_wifi_logging()
 #endif // WIFI_LOG_SUBMODULE_ALL
 }
 
+void Esp32WiFiManager::start_ssid_scan(Notifiable *n)
+{
+    OSMutexLock l(&ssidScanResultsLock_);
+    ssidScanResults_.clear();
+    ssidCompleteNotifiable_ = n;
+    wifi_scan_config_t cfg;
+    bzero(&cfg, sizeof(wifi_scan_config_t)); // active scan all channels, 120ms per channel
+    ESP_ERROR_CHECK(esp_wifi_scan_start(&cfg, false));
+}
+
+size_t Esp32WiFiManager::get_ssid_scan_result_count()
+{
+    OSMutexLock l(&ssidScanResultsLock_);
+    return ssidScanResults_.size();
+}
+
+wifi_ap_record_t Esp32WiFiManager::get_ssid_scan_result(size_t index)
```

Comment:
const wifiap_record_t& as return

Reply:
done

Reply:
sorry this was a wrong advice. it does need to be returned by value.